### PR TITLE
Add monitoring/grafana_folder module

### PR DESCRIPTION
**Pull request by martinwangjian**
##### SUMMARY

**grafana_folder** is a new Ansible module that allow you to create/update/delete grafana folder

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

grafana_folder

##### ANSIBLE VERSION

```
ansible 2.7.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.6 (default, Jul 19 2018, 16:29:00) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

##### ADDITIONAL INFORMATION

This module can manage grafana folder. Folder is a feature of grafana 5.
I've tested cases:

```paste below
- hosts: localhost
  tasks:
    - name: create folder a
      grafana_folder:
        grafana_url: http://grafana.company.com
        grafana_api_key: "{{ grafana_api_key }}"
        state: present
        uid: "a"
        title: "a"

   - name: create folder b
     grafana_folder:
       grafana_url: http://grafana.company.com
       state: present
       uid: "b"
       title: "b"

   - name: create folder c
     grafana_folder:
       grafana_url: http://grafana.company.com
       state: present
       uid: "c"
       title: "c"

    - name: delete folder b
      grafana_folder:
        grafana_url: http://grafana.company.com
        grafana_api_key: "{{ grafana_api_key }}"
        state: absent
        uid: b

    - name: update folder
      grafana_folder:
        grafana_url: http://grafana.company.com
        grafana_user: "admin"
        grafana_password: "{{ grafana_password }}"
        state: present
        uid: "c"
        title: "d"

    - name: update folder
      grafana_folder:
        grafana_url: http://grafana.company.com
        grafana_user: "admin"
        grafana_password: "{{ grafana_password }}"
        state: present
        uid: "c"
        title: "e"
        overwrite: yes
```

### CODE REVIEW

File: lib/ansible/modules/monitoring/grafana_folder.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+      - Not used when I(grafana_api_key) is set, because the grafana_api_key only belongs to one organisation..
+    default: 1
+  state:
+    description:
+      - State of the folder.
+    required: true
+    choices: [ absent, present ]
+    default: present
+  uid:
+    required: true
```
Comment (seuf) [inline lib/ansible/modules/monitoring/grafana_folder.py]: according to the grafana API documentation, the uid is optional.

**Review by seuf — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/monitoring/grafana_folder.py (hunk 1 above)
Comment (martinwangjian) [reply lib/ansible/modules/monitoring/grafana_folder.py]: thx, fixed

**Comment by seuf** [comment]
I've tried to create a new Folder with a space inside the title and it failed :
`fatal: [localhost]: FAILED! => {"changed": false, "msg": "error : list indices must be integers or slices, not str"}`

**Comment by rrey** [comment]
Hi @martinwangjian,

Are you still interested in making the required changes in this PR ?
I'm ready to help !

**Comment by rrey** [comment]
close_me

**Pull request closed without merging**